In [1]:
import scipy.stats as stats
import pandas as pd
import numpy as np

In [2]:
data = [100, 150, 50, 100, 130, 120, 100, 110, 75, 65, 150, 120, 50, 100, 100, 140, 90, 150, 50, 90]

## Define H0 & H1

##### H0 -> Mean of Transaction amount = 100
##### H1 -> Mean of Transaction amount ≠ 100

In [3]:
stats.ttest_1samp(a = data, popmean = 100)

Ttest_1sampResult(statistic=0.27316577116663854, pvalue=0.7876728941695149)

##### T-test statistics = 0.2731657711666385
##### P-Value = 0.787672894169515

Jika menggunakan alpha = 5%, maka kesimpulan yang bisa ditarik adalak P-Value > alpha, yang berarti kita gagal untuk menolak H0

Conclusion -> **The mean of Transaction amount is equal to 100**

Namun bila ingin menggunakan niali t-statistic, maka perlu dicari nilai critical region terlebih dahulu

## Two Tailed Test

In [4]:
# Two tailed test

stats.t.ppf(1-(0.05/2), 19)

2.093024054408263

##### Critical region: t < -2.093 atau t > 2.093
Karena nilai t-statistic tidak berada pada critical region, maka kita gagal untuk menolak H0.

Conclusion -> **The mean of Transaction amount is equal to 100**



## One Tailed Test

In [5]:
# One tailed test

0.787672894169515/2

0.3938364470847575

Jika menggunakan alpha = 5%, maka kesimpulan yang bisa ditarik adalak P-Value/2 > alpha, yang berarti kita gagal untuk menolak H0

Conclusion -> The mean of Transaction amount is equal to 100

In [6]:
# H1 -> Mean of Transaction amount > 100 (Right side)

stats.t.ppf(1-0.05, 19)

1.729132811521367

##### Critical Region: t > 1.729
Karena nilai t-statistic tidak berada pada critical region, maka kita gagal untuk menolak H0.

Conclusion -> The mean of Transaction amount is equal to 100

In [7]:
# H1 -> Mean of Transaction amount < 100 (Left side)

stats.t.ppf(0.05, 19)

-1.7291328115213678

##### Critical Region: t < -1.729
Karena nilai t-statistic tidak berada pada critical region, maka kita gagal untuk menolak H0.

Conclusion -> The mean of Transaction amount is equal to 100